### PR TITLE
Improvement/cal 639 turn edit location dialog into radix uu

### PR DESCRIPTION
**Pull request by alishaz-polymath**
- Removes headless UI from event-types/[type].tsx
- Replaces Modal with Dialog in Edit Location
- Adds react-hook-form 
- Fixes Radio button bug

### CODE REVIEW

**Review by emrysal — APPROVED** [review]
Font size on period types is a bit too big, minor thing:

image

I've managed to reproduce the bug with multiple locations. It happens when you open the dialog, then cancel, then drop the select again, cancel.. rince repeat and you'll break it more and more. Up to you if you want to take a look or have a look (or someone else) in a dedicated PR.

image

Code itself is a big improvement, well done

**Comment by alishaz-polymath** [comment]
>I've managed to reproduce the bug with multiple locations

Yeah, that bug is frustrating. I haven't been able to pinpoint the reason behind it. It's in prod, and I couldn't figure out why this is, hence not really worked on it in this PR.

**Review by zomars — COMMENTED** [review]
Thank you for doing this! I left some comments about minor details.

File: pages/event-types/[type].tsx [hunk 1]
```diff
@@ -590,10 +602,19 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
-                        value={asStringOrUndefined(eventType.schedulingType)}
-                        options={schedulingTypeOptions}
+                        control={formMethods.control}
+                        defaultValue={eventType.schedulingType}
+                        render={() => (
+                          <RadioArea.Select
+                            value={asStringOrUndefined(eventType.schedulingType)}
+                            options={schedulingTypeOptions}
+                            onChange={(val) => {
+                              formMethods.setValue("schedulingType", val);
```
[the remark this answers is not in the document]
Comment (alishaz-polymath) [reply pages/event-types/[type].tsx:614]: This one is actually related to the `components/ui/form/radio-area/RadioAreaGroup` definition of onChange. I'm not sure what's the best way to proceed in this case.

File: pages/event-types/[type].tsx [hunk 2]
```diff
@@ -68,15 +63,15 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
   const { t } = useLocale();
   const PERIOD_TYPES = [
     {
-      type: "rolling",
+      type: "ROLLING",
```
[the remark this answers is not in the document]
Comment (alishaz-polymath) [reply pages/event-types/[type].tsx]: Thanks for this. I was a bit unsure about it.

File: pages/event-types/[type].tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (zomars) [reply pages/event-types/[type].tsx:614]: I think we make this in a follow up PR. It's good enough for now 👍🏽

**Review by zomars — APPROVED** [review]

**Pull request merged**